lib/discourse_event.rb

```diff
@@ -21,6 +21,16 @@ def self.on(event_name, &block)
         raise_error: true,
       )
     end
+
+    if event_name == :user_badge_removed
+      Discourse.deprecate(
+        "The :user_badge_removed event is deprecated. Please use :user_badge_revoked instead",
+        since: "3.1.0beta4",
+        drop_from: "3.2",
```

Comment:
Beta4 has already been released (and didn't contain this deprecation) 😛 
And I don't think we need to wait ~2 stable releases until removal?

```suggestion
        since: "3.1.0beta5",
        drop_from: "3.2.0.beta1",
```